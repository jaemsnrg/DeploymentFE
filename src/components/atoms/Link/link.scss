@import "~styles/_mixins.scss";
@import "~styles/_colours.scss";
@import "~styles/_variables.scss";

.action__link:hover {
  text-decoration: underline !important;
  cursor: pointer;
}

.action__link {
  color: $blue-dark;
  &--large {

  }
}

