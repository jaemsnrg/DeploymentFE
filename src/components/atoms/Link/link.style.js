import styled from 'styled-components';

export const LinkWrapper = styled.a`
  text-decoration: none;
  color: ${({ theme, color }) => color ? theme.colours[color] : theme.colours.blueDark};
  ${({largeLink, mediumLink, medLink, theme}) => {
    try {
      if (largeLink) {
        return theme.text.bxlMedium
      } else if (mediumLink) {
        return theme.text.bm
      } else if (medLink) {
        return theme.text.bmMedium
      } else {
        return theme.text.bsMedium
      }
    } catch (error) {}
  }};
  &:hover {
    text-decoration: underline !important;
    cursor: pointer;
  }

`