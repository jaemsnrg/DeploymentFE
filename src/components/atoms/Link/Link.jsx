import React from 'react';
import PropTypes from 'prop-types';
import { LinkWrapper } from './link.style';

const Link = (props) => {
  const onClick = (event) => {
    if (props.onClick) {
      props.onClick(event);
    }
  };

  return (
    <LinkWrapper
      largeLink={props.largeLink}
      mediumLink={props.mediumLink}
      medLink={props.medLink}
      href={props.linkLocation}
      onClick={onClick}
      color={props.color}
      
    >
      {props.linkTitle}
    </LinkWrapper>
  );
};

Link.propTypes = {
  mediumLink: PropTypes.bool,
  largeLink: PropTypes.bool,
  linkLocation: PropTypes.string,
  linkTitle: PropTypes.string,
  onClick: PropTypes.func,
  color: PropTypes.string,
};

export default Link;
