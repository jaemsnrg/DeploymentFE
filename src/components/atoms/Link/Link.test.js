import React from "react";
import renderer from "react-test-renderer";

import Link from "./Link";

describe("Link component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Link />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
