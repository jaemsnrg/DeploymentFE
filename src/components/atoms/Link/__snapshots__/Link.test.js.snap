// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component matches the snapshot 1`] = `
<a
  className="function link() { [native code] }  "
  onClick={[Function]}
/>
`;
